docs/api/io/BufferedSerial.md

```diff
@@ -0,0 +1,58 @@
+
+TX and RX pins can be specified as Not Connected (NC) for simplex (unidirectional) communication or as valid pins for full duplex (bi-directional) communication.
+
+Data is sent/received at a pre-defined speed called baud rate. Standard baud rate include 9600, 119200, 115200 or others.
+
+Pieces of data are transmitted using data packets of configurable sizes divided in different sections which include:
+
+* Start bit: Indicates the start of UART data transmission
+* Data frame: Can be 5 to 8 (or 9 if a parity bit is not used) bits long for the actual data being transferred.
+* Parity bit: Optional bit, used for data error detection.
+* Stop bits: Can be 1 to 2 bits long to signal the end of a data packet.
+
+The `BufferedSerial` class maps targets underlying serial communication functions. See the [porting guide](../porting/serial-port.html) for target serial support.
+
+When the RX interrupt is trigged, the `BufferedSerial` class stores the byte(s) available to read from the hardware buffer to an internal intermediary buffer. When a read request is made, the `BufferedSerial` class uses a mutex lock and enters a critical section to read out the number of bytes requested if as many are available in the intermediary buffer.
+To transmit multiple bytes, the class uses an intermediary buffer to store the bytes to send and monitors the serial interface to transfer them to the hardware buffer as soon as it is available. However, all bytes are written unbuffered if in a critical section.
+Using intermediary buffers allows it to be used reliably for input from non-interrupt context whilst avoiding excess spinning waiting for transmission buffer space.
+
+## Configuration
+
+The following parameters can be configured at object instantiation:
+
+  - _TX_
+  - _RX_
+  - _Baud rate_ - defaults to the value configured in `mbed-os/platform/mbed_lib.json`.
+
+The following parameters can be configured after an `BufferedSerial` object instantiation.
+
+  - _Baud rate_
+  - _Data frame length_
+  - _Parity bit_
+  - _Stop bits_
+
+The default settings for a microcontroller are described as _9600-8-N-1_, a common notation for serial port settings.
+
+Additionally, hardware flow control can also be configured if necessary.
+
+You can view more information about the configurable settings and functions in the class reference.
+
+## Class reference
+
+[![View code](https://www.mbed.com/embed/?type=library)](https://os.mbed.com/docs/mbed-os/development/mbed-os-api-doxy/classmbed_1_1_buffered_serial.html)
+
+<span class="notes">**Note**: On a Windows machine, you need to install a USB serial driver. See [Windows serial configuration](../tutorials/serial-communication.html#windows-serial-driver).</span>
+
+## Example
+
+[![View code](https://www.mbed.com/embed/?url=https://os.mbed.com/teams/mbed_example/code/BufferedSerial/)](https://os.mbed.com/teams/mbed_example/code/BufferedSerial/file/112a40a5991a/main.cpp)
```

Comment:
It would be good to provide multiple snippets like Serial and RawSerial did.
Perhaps you could comment specifically on printing using `printf` and show code snippet since there were many previsus examples on how to do it from a serial class.

Reply:
https://github.com/ARMmbed/mbed-os-5-docs/compare/b14e6f622f7fbb29482b769636c4c4ab4b513ee5..934953559ca74a40ae8258e144792d1f2efc79af